Framework/Core/src/DataRelayer.cxx

```diff
@@ -512,9 +512,10 @@ void DataRelayer::getReadyToProcess(std::vector<DataRelayer::RecordAction>& comp
         auto payload = partial[idx].payload(part).get();
         return DataRef{nullptr,
                        reinterpret_cast<const char*>(header->GetData()),
-                       reinterpret_cast<const char*>(payload ? payload->GetData() : nullptr)};
+                       reinterpret_cast<char const*>(payload ? payload->GetData() : nullptr),
+                       payload ? payload->GetSize() : 0};
       }
-      return DataRef{nullptr, nullptr, nullptr};
+      return DataRef{nullptr, nullptr, nullptr, 0};
```

Comment:
I think this can now be `return {};` which will keep clang-code-checker happy (there is a modernize check which fails, when enabled).

Reply:
ok, fixed